### ISSUE TITLE
Fix Firefox extension to work with a server hosted on not top-level domain

### ISSUE THREAD

**Issue by abhin4v** [wezm/feedlynx-ext#3]
The current Firefox extension code assumes that the Feedlynx server is accessible at some top-level domain, for example: https://feedlynx.example.com. If I configure my server to be accessible at a path such as https://example.com/feedlynx, the current extension code is not able to submit to it. This is because of this code in background.ts/js:

```js
const post_url = new URL("/add", results.host);
```

Replacing it with this code allows my use-case to work:

```js
const post_url = new URL("add", results.host.endsWith('/') ? results.host : results.host + '/');
````

Note that the iOS shortcut does not have this behaviour and my use-case works fine with it.

**Comment by wezm** [comment]
I did assume that Feedlynx would be deployed to its own domain, although I see I now that I did not document this. I can also understand that it's not clear that the configuration of the extension expects only a scheme and domain. Originally it was just the domain but I needed to allow specifying http or https during development.

Running under a prefix can get awkward if the application generates redirects (or other references to itself) since it needs to know to include the prefix when generating the URL. It works for now because Feedlynx does not currently generate any redirects.

I'll apply this change now but I can't guarantee that this form of deployment will always work.
